src/main/api/catalog.ts

```diff
@@ -71,64 +71,63 @@ export class CatalogApi implements ICatalogApi {
         const {
             audio: {
                 readed: audiobookReaded,
-                added: audiobookAdded,
             },
             divina: {
                 readed: divinaReaded,
-                added: divinaAdded,
             },
             epub: {
                 readed: epubReaded,
-                added: epubAdded,
             },
             pdf: {
                 readed: pdfReaded,
-                added: pdfAdded,
+            },
+            all: {
+                added: allAdded,
             },
         } = await this.getPublicationView();
 
         // Dynamic entries
         const entries: CatalogEntryView[] = [
-            {
-                title: __("catalog.entry.continueReading"),
-                totalCount: epubReaded.length,
-                publicationViews: epubReaded,
-            },
             {
                 title: __("catalog.entry.lastAdditions"),
-                totalCount: epubAdded.length,
-                publicationViews: epubAdded,
+                totalCount: allAdded.slice(0, NB_PUB).length,
+                publicationViews: allAdded.slice(0, NB_PUB),
```

Comment:
`allAdded.slice(0, NB_PUB)` has a cost (array allocation + shallow copy, i.e. clone of primitive types, or pointer copies for structured objects).
Therefore, I would advice not invoking `slice()` for both `totalCount` and `publicationViews`.

PS: same remark for all other cases in this PR.

Technical references:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer/slice

Note the difference in NodeJS between `TypedArray#slice()`, `Buffer#slice()` and `TypedArray#subarray()`:

https://nodejs.org/api/buffer.html#buffer_buffers_and_typedarrays

> While TypedArray#slice() creates a copy of part of the TypedArray, Buffer#slice() creates a view over the existing Buffer without copying. This behavior can be surprising, and only exists for legacy compatibility. TypedArray#subarray() can be used to achieve the behavior of Buffer#slice() on both Buffers and other TypedArrays.

Reply:
thanks for the resources 👍

Reply:
Fixed via https://github.com/edrlab/thorium-reader/pull/1292/commits/f3038a774bd3f76383d805e38342fd6350fd2fd7